src/js/brim/ingest/fileList.ts

```diff
@@ -1,46 +1,52 @@
 import {IngestFileType} from "./detectFileType"
 import lib from "../../lib"
 
-export type FileListData = {type: IngestFileType; path: string}[]
+export type FileListData = {type: IngestFileType; file: File}[]
```

Comment:
not sure if you considered extending the `File` interface by annotating it with the new `type` field? would save all the extra `.file.`s since it would be flattened, just a thought!

Reply:
Yeah true, I usually try to stay away from extending things like that. "Compose rather than inherit" mindset I guess. Is this something that is done conventionally in go?

Reply:
I think the composition vs inheritance dilemma is the same in go. I do agree with the compose > inheritance mindset, but thought that in this case with just these two types, and one always depending on the other, extending would grant the small bonus of less typing as you access the imbedded File interface. I have no strong feelings in this case at all though!